operator/v1/types_network.go

```diff
@@ -562,6 +562,12 @@ type IPsecConfig struct {
 	// +kubebuilder:validation:Enum=Disabled;External;Full
 	// +optional
 	Mode IPsecMode `json:"mode,omitempty"`
+	// encapsulation uses rfc3949 encapsulation of IKE and IPsec encryption packets
+	// This is applicable only when mode is set to Full.
+	// Default is false.
+	// +optional
+	// +kubebuilder:default:=false
+	Encapsulation bool `json:"encapsulation,omitempty"`
```

Comment:
Bools are no longer permitted in our APIs, they do not evolve well, review our API conventions for further explanation.

You should come up with a meaningful enum instead in most cases